src/midifunctor.cpp

```diff
@@ -336,16 +344,186 @@ FunctorCode InitMIDIFunctor::VisitArpeg(const Arpeg *arpeg)
     });
 
     // Defer the notes in playing order
-    double shift = 0.0;
-    const double increment = UNACC_GRACENOTE_DUR * m_currentTempo / 60000.0;
-    for (const Note *note : sortedNotes) {
-        if (shift > 0.0) m_deferredNotes[note] = shift;
-        shift += increment;
+    Fraction shift = 0;
+    Fraction startTime = sortedNotes.front()->GetScoreTimeOffset();
```

Comment:
@brdvd I think this is the issue:
```suggestion
    Fraction startTime = sortedNotes.front()->GetScoreTimeOnset();
```

Reply:
Yes indeed!

Reply:
With the suggested fix, this MR should be ready to merge.